state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -708,20 +724,10 @@ pub mod pallet {
 						debug_assert!(false, "Swap is not completed yet!");
 					}
 				}
+
+				FirstBlockWithPendingSwaps::<T>::set(block + 1u32.into());
```

Comment:
nit: The block number in `FirstBlockWithPendingSwaps` doesn't necessarily contain any swaps so the name is a bit misleading. I would suggest storing a `LastBlockExecuted` or similar. Then we don't rely on having matching logic in the function and outside (ie. we increment the block here, and also in the loop where we increment `block_to_process`).

In fact we can write to storage outside the loop:

```rust
let mut block_to_execute = LastBlockExecuted::get();
while block_to_execute <= current_block - 2 {
	match process_swaps_for_block(block_to_execute) {
		Ok(()) => block_to_execute += 1,
		Err(_) => break;
	}
}
LastBlockExecuted::put(block_number);
```

Reply:
> I would suggest storing a LastBlockExecuted or similar.

That was my first thought too, but then I realised that it would be awkward around block 0, since I would need to use None or something like that to indicate that block 0 hasn't been processed. Of course, in practise we could just initialize it to 0 anyway, but I wanted to be more precise and went for a different name instead.

> The block number in FirstBlockWithPendingSwaps doesn't necessarily contain any swaps so the name is a bit misleading.

How about `FirstUnprocessedBlock`?

> In fact we can write to storage outside the loop

My intention was to write block number as part of the "transaction". I know `LastBlockExecuted::put(block_number)` probably can't fail, but conceptually it belongs there. I don't think it is much of an optimisation either since usually we would be processing 1 block only.

Reply:
Update: I renamed `FirstBlockWithPendingSwaps` -> `FirstUnprocessedBlock`, and moved `FirstUnprocessedBlock::<T>::set(block_to_process);` outside of `process_swaps_for_block` (and outside of the loop). Let me know if you still feel strongly about calling the `LastBlockExecuted` naming after my arguments against it.

Reply:
Naming is fine 👌